### PR TITLE
Feat-630 - Redesign of Date Range Picker

### LINKED ISSUE #630: [Frontend] Redesign date range picker
### User Story

As a user, I want an intuitive interface to select date ranges, so that I can properly filter the dataset to retrieve the information I need.

### Descriptions

Some people have expressed difficulty with the UX of our current date picker. This ticket attempts to create a new UI for the date range picker to propose as alternative to Design and Product.

This new design comes from various data dashboards. In the attached examples, these come from AWS CloudWatch.

You will still need to adhere to our design system, but the layout and experience itself will be different.

First, you will need to focus on the main UI for this, namely the two calendars side-by-side, the boxes for datatimes, etc. Afterwards, we can discuss on how to include presets. I have few ideas.

### Acceptance Criteria

- [ ] Build a new, alternative date range picker
- [ ] Present to Design and Product

### Design References

Image
Image

Relative Presets:

- Days - 1, 2, 3, 4, 5, 6
- Weeks - 1, 2, 3, 4
- Months - 3, 6, 9, 12

### PR DESCRIPTION
**Pull request by gibsonliketheguitar**
### Description

Redesign Implementation of calendar month.
- Calendar app code was cannibalize and 2 month calendar was created along with utility functions
- Added suggestion tab to allow user to select presets

Notes
- Might have read instruction wrong, but I can make changes to relative preset fairly easily

### Related Issues

Resolves #630 

### Testing

- Ran testing suites.
```
  Date Range 2 month Calendar
    ✓ renders Back Arrow Icon (45 ms)
    ✓ renders Forward Arrow Icon (9 ms)
    ✓ renders current month and next month (14 ms)
    ✓ render current month's year and next month's year (13 ms)
  Date Range Call to Action 
    ✓ renders current month and next month (19 ms)
    ✓ render December 2024 and January 2025 when Forward Arrow is pressed (224 ms)
    ✓ render October 2024 and November 2024 when Forward Arrow is pressed (52 ms)
    
    Date Range Suggestions Radio Group
    ✓ renders radio group and all it's options (17 ms)
```

### Checklist

- [x] I have followed all conventions outlined in our documentation.
- [ ] I have fully tested my changes and confirmed that all new and existing tests pass.
- [x] I have written meaningful commit messages for all changes.
- [x] I have linted and formatted my changes to follow the code style of this repository.
- [x] I have updated our documentation, accordingly.
- [x] I have checked currently opened pull requests to ensure that there are no pending pull request for the same changes or issues.
- [x] I have confirmed that this pull request fully meets the acceptance criteria for all related issues listed above.

### CODE REVIEW

File: packages/ui/src/components/date-picker/date-picker-calendar.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        </div>
+      </div>
+
+      {/** Actual Calendars */}
+      <div className="flex w-full">
+        <div className="flex justify-center px-4 pb-2">
+          <table className="border-collaspse">
+            <thead>
+              <tr>
+                {["M", "T", "W", "T", "F", "S", "S"].map((ele: string, index: number) => (
```
Comment (glenflorendo) [inline packages/ui/src/components/date-picker/date-picker-calendar.tsx:104]: The week should start with "S" for "Sunday".

File: packages/ui/src/components/date-picker/date-picker-calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  return (
+    <div data-testid="date-range-calendar" className="bg-white-100 w-[512px]">
+      {/**Calendar Month Nav */}
+      <div className="flex h-[52px] flex-1 justify-between px-4">
+        <div className="flex flex-1 items-center justify-start space-x-4">
+          <button aria-label="prev-months" className="px-2" onClick={() => handleUpdateMonth("prev")}>
+            <ArrowBackIosNewIcon sx={{ fontSize: 8, color: "#7A7A7B" }} />
+          </button>
+          <div className="space-x-1">
+            <span aria-label="first-date-month"> {MONTH_NAMES[month]} </span>
```
Comment (glenflorendo) [inline packages/ui/src/components/date-picker/date-picker-calendar.tsx:82]: The calendar titles should be centered aligned with the calendar.

File: packages/ui/src/components/date-picker/date-picker-suggestions.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import RadioGroup from "../radio-group";
+
+interface DatePickerSuggestion {
+  onSuggestionChange: (arg0: string) => void;
+}
+
+export enum RelativeDatePresets {
+  "YEARS_1" = "Past 1 Year",
+  "YTD" = "Year to Date",
+  "MONTHS_3" = "3 Months",
```
Comment (glenflorendo) [inline packages/ui/src/components/date-picker/date-picker-suggestions.tsx:11]: This should be Past 3 Months.

File: packages/ui/src/components/date-picker/date-picker-suggestions.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  "MONTHS_1" = "1 Month",
```
Comment (glenflorendo) [inline packages/ui/src/components/date-picker/date-picker-suggestions.tsx:12]: This should be past 1 Month.

**Review by glenflorendo — CHANGES_REQUESTED** [review]
Solid work. There are just a few things to fix, then it should be all ready to merge!

**Comment by gibsonliketheguitar** [comment]
Feeling nervous about the force push. Should I roll it back?

**Pull request merged**